types/mempool/sender_nonce_property_test.go

```diff
@@ -0,0 +1,117 @@
+package mempool_test
+
+import (
+	"math/rand"
+	"sort"
+
+	"pgregory.net/rapid"
+
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	mempool "github.com/cosmos/cosmos-sdk/types/mempool"
+	simtypes "github.com/cosmos/cosmos-sdk/types/simulation"
+	"github.com/cosmos/cosmos-sdk/x/auth/signing"
+	"github.com/stretchr/testify/require"
+	"github.com/tendermint/tendermint/libs/log"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+)
+
+var (
+	_ sdk.Tx                  = (*testTx)(nil)
+	_ signing.SigVerifiableTx = (*testTx)(nil)
+	_ cryptotypes.PubKey      = (*testPubKey)(nil)
+)
+
+// Property Based Testing
+// Split the senders tx in independent slices and then test the following properties in each slice
+// same elements input output and overwrite the nonce duplicates
+// for every sequence element pair a, b a < b
+
+var genAddress = rapid.Custom(func(t *rapid.T) simtypes.Account {
+	accounts := simtypes.RandomAccounts(rand.New(rand.NewSource(rapid.Int64().Draw(t, "seed for account"))), 1)
```

Comment:
We'll always get the same accounts generated since the seed source is "seed for account" right?